apps/ledger-live-mobile/src/screens/CustomImage/PreviewPostEdit.tsx

```diff
@@ -0,0 +1,151 @@
+import FramedImage from "../../components/CustomImage/FramedImage";
+import ImageProcessor, {
+  Props as ImageProcessorProps,
+  ProcessorPreviewResult,
+  ProcessorRawResult,
+} from "../../components/CustomImage/ImageProcessor";
+
+type NavigationProps = BaseComposite<
+  StackNavigatorProps<
+    CustomImageNavigatorParamList,
+    ScreenName.CustomImagePreviewPostEdit
+  >
+>;
+
+const PreviewPostEdit = ({ navigation, route }: NavigationProps) => {
+  const { t } = useTranslation();
+  const { params } = route;
+  const { baseImageFile, contrast, imagePreview, device } = params;
+
+  const handleError = useCallback(
+    (error: Error) => {
+      console.error(error);
+      navigation.navigate(ScreenName.CustomImageErrorScreen, { error, device });
+    },
+    [navigation, device],
+  );
+
+  const [previewLoading, setPreviewLoading] = useState<boolean>(true);
+  const [processorPreviewImage, setProcessorPreviewImage] =
+    useState<ProcessorPreviewResult | null>(null);
+  const [rawResultLoading, setRawResultLoading] = useState(false);
+  const imageProcessorRef = useRef<ImageProcessor>(null);
+
+  const handlePreviewResult: ImageProcessorProps["onPreviewResult"] =
+    useCallback(
+      data => {
+        setProcessorPreviewImage(data);
+        setPreviewLoading(false);
+      },
+      [setProcessorPreviewImage],
+    );
+
+  const handleRawResult: ImageProcessorProps["onRawResult"] = useCallback(
+    (data: ProcessorRawResult) => {
+      if (!processorPreviewImage) {
+        /**
+         * this should not happen as the "request raw result" button is only
+         * visible once the preview is there
+         * */
+        throw new ImagePreviewError();
+      }
+      navigation.navigate(ScreenName.CustomImageStep3Transfer, {
+        rawData: data,
+        previewData: processorPreviewImage,
+        device,
+      });
+      setRawResultLoading(false);
+    },
+    [navigation, setRawResultLoading, processorPreviewImage, device],
+  );
+
+  const handlePreviewImageError = useCallback(
+    ({ nativeEvent }: NativeSyntheticEvent<ImageErrorEventData>) => {
+      console.error(nativeEvent.error);
+      handleError(new ImagePreviewError());
+    },
+    [handleError],
+  );
+
+  const requestRawResult = useCallback(() => {
+    imageProcessorRef?.current?.requestRawResult();
+    setRawResultLoading(true);
+  }, [imageProcessorRef, setRawResultLoading]);
+
+  const handleEditPicture = useCallback(() => {
+    navigation.navigate(NavigatorName.CustomImage, {
+      screen: ScreenName.CustomImageStep1Crop,
+      params: {
+        device,
+        baseImageFile,
+        isPictureFromGallery: false,
+      },
+    });
+  }, [navigation, device, baseImageFile]);
+
+  if (!imagePreview) {
+    return <InfiniteLoader />;
+  }
+
+  return (
+    <SafeAreaView style={{ flex: 1 }} edges={["bottom"]}>
+      <Flex flex={1}>
+        <Flex flex={1}>
+          {imagePreview?.imageBase64DataUri && (
+            <ImageProcessor
```

Comment:
Would it be possible to remove this processing logic completely from this step as it's already handled in the previous screen `Step2Preview` and thus we are doing the same processing twice in a row (you just need to put back the logic to get the raw result there).
In this screen then you would completely eliminate any image processing logic and get the `ProcessingPreviewResult` and `ProcessingRawResult` as route params.

Reply:
Done